src/main/java/exercise39/SortingRecords.java

```diff
@@ -60,8 +60,11 @@ public String recordSorter(List<Map<String, String>> records) {
     public String recordFilter(String searchTerm, List<Map<String, String>> records) {
 
         String filteredRecords = "";
+        //removeIf?
         for (Map<String, String> map : records) {
+            //can we do some of the "challenges"?
             if (map.get("First Name").contains(searchTerm) || map.get("Last Name").contains(searchTerm)) {
+                //repetition?
```

Comment:
We've been a bit cheeky here and ignored the complication around the date field.  It does highlight though that we're repeating the code that does the string building, can you think of a way to fix the date issue and reduce the duplication?